src/wasm/working.cc

```diff
@@ -0,0 +1,115 @@
+#include <emscripten.h>
+#include <stdlib.h>
+#include <string_view>
+#include <string>
+#include <map>
+
+EM_JS(void, console, (const char* msg), {})
+EM_JS(void, consoleN, (const char* msg, int num), {})
+
+#define MKSTR(text, args)  (std::string(text) + args).c_str()
+
+
+class Handle;
+
+EM_JS(void, handleSet, (Handle* handle, int num), {})
+
+
+class Handle {
+public:
+  Handle(const char* name) : name_(name), num_(0) {}
```

Comment:
Does the compiler not handle name shadowing properly here?

Reply:
It does, but the underscore is to mark member fields, not to (unnecessarily) avoid shadowing.